cmake/Modules/FindQuadmath.cmake

```diff
@@ -11,24 +11,34 @@ include(CheckCXXSourceCompiles)
 include(CMakePushCheckState)
 include(CheckCXXCompilerFlag)
 
-check_cxx_compiler_flag("-Werror -fext-numeric-literals" HAVE_EXTENDED_NUMERIC_LITERALS)
-if (HAVE_EXTENDED_NUMERIC_LITERALS)
+if(NOT DEFINED USE_QUADMATH OR USE_QUADMATH)
+  if(NOT DEFINED HAVE_EXTENDED_NUMERIC_LITERALS)
+    check_cxx_compiler_flag("-Werror -fext-numeric-literals" HAVE_EXTENDED_NUMERIC_LITERALS)
+  endif()
+
+  if (HAVE_EXTENDED_NUMERIC_LITERALS)
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fext-numeric-literals")
-endif()
+  endif()
 
-cmake_push_check_state()
-list(APPEND CMAKE_REQUIRED_LIBRARIES "quadmath")
-CHECK_CXX_SOURCE_COMPILES("
+  cmake_push_check_state()
+  list(APPEND CMAKE_REQUIRED_LIBRARIES "quadmath")
```

Comment:
ah, my bad. I think I finally got what you meant. I updated the patch. is it better now?